pkg/controller/instance/instance_controller.go

```diff
@@ -265,49 +267,39 @@ func instanceEventFilter(mgr manager.Manager) predicate.Funcs {
 					}
 				}
 
-				if err = createPlan(mgr, planName, new); err != nil {
+				if err = createPlanOld(mgr, planName, new); err != nil {
 					log.Printf("InstanceController: Error creating PlanExecution \"%v\" for instance \"%v\": %v", planName, new.Name, err)
 				}
 			}
 
 			// See if there's a current plan being run, if so "cancel" the plan run.
 			return e.ObjectOld != e.ObjectNew
 		},
-		// New Instances should have Deploy called
+		// New Instances should confirm there is a deployment plan without side-effects)
 		CreateFunc: func(e event.CreateEvent) bool {
 			log.Printf("InstanceController: Received create event for instance \"%v\"", e.Meta.GetName())
+			ctx := context.TODO()
+
+			// In order to create we must have an OV and it must have a "deploy" plan.
 			instance := e.Object.(*kudov1alpha1.Instance)
 
-			// Get the instance OperatorVersion object
-			ov := &kudov1alpha1.OperatorVersion{}
-			err := mgr.GetClient().Get(context.TODO(),
-				types.NamespacedName{
-					Name:      instance.Spec.OperatorVersion.Name,
-					Namespace: instance.GetOperatorVersionNamespace(),
-				},
-				ov)
-			if err != nil {
+			ov, err := getOperatorVersion(ctx, mgr.GetClient(), nil, instance)
+			if err != nil || ov == nil {
 				log.Printf("InstanceController: Error getting operatorversion \"%v\" for instance \"%v\": %v",
 					instance.Spec.OperatorVersion.Name,
 					instance.Name,
 					err)
 				// TODO: We probably want to handle this differently and mark this instance as unhealthy
-				// since its linking to a bad OV.
+				// no ov, no create of instance
 				return false
 			}
 
 			planName := "deploy"
```

Comment:
I feel like this is something we'll be doing inside reconcile as well, I would keep it in one place, I don't think it's optimization to do it here as well as there. At least all this logic will be in one place

Reply:
same as comment above... any side effects have been removed...  but we are checking to see if there is an OV and a deploy plan... otherwise we do not process the create event.